merkledag.go

```diff
@@ -196,14 +191,14 @@ func FetchGraphWithDepthLimit(ctx context.Context, root cid.Cid, depthLim int, s
 		return false
 	}
 
-	v, _ := ctx.Value(progressContextKey).(*ProgressTracker)
-	if v == nil {
+	progressTracker := GetProgressTracker(ctx)
+	if progressTracker == nil {
 		return EnumerateChildrenAsyncDepth(ctx, GetLinksDirect(ng), root, 0, visit)
 	}
 
 	visitProgress := func(c cid.Cid, depth int) bool {
 		if visit(c, depth) {
-			v.Increment()
+			progressTracker.PlanToPin(c)
```

Comment:
Out of curiosity, by the time this line is running have we already started pinning the cid? If so, should we call it `Pinning(cid)` maybe? I actually don't know, just curious what state to communicate here.